### PR TITLE
James 1675

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    private Consumer<? super MessageId> delete(MessageMapper<Id> messageMapper, MailboxSession mailboxSession, SetMessagesResponse.Builder responseBuilder) {
+        return (messageId) -> {
+            try {
+                Mailbox<Id> mailbox = mailboxMapperFactory
+                        .getMailboxMapper(mailboxSession)
+                        .findMailboxByPath(messageId.getMailboxPath(mailboxSession));
+
+                Iterator<MailboxMessage<Id>> mailboxMessage = messageMapper.findInMailbox(mailbox, MessageRange.one(messageId.getUid()), FetchType.Metadata, LIMIT_BY_ONE);
```
Comment (mbaechler) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java]: now you can extract these 4 lines into a `get` method

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/json/ParsingWritingObjectsTest.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+import com.fasterxml.jackson.databind.ser.impl.SimpleBeanPropertyFilter;
+import com.fasterxml.jackson.databind.ser.impl.SimpleFilterProvider;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+public class ParsingWritingObjectsTest {
+
+
+    private static final MessageId MESSAGE_ID = MessageId.of("username|mailbox|1");
```
Comment (mbaechler) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/json/ParsingWritingObjectsTest.java]: using an interface for your fixture has several advantages : 
* you can name your fixture
* you can drop private static final

File: server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java (hunk 1 above)
Comment (aduprat) [inline server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMessagesMethod.java]: OK

File: server/protocols/jmap/src/test/java/org/apache/james/jmap/json/ParsingWritingObjectsTest.java (hunk 2 above)
Comment (aduprat) [inline server/protocols/jmap/src/test/java/org/apache/james/jmap/json/ParsingWritingObjectsTest.java]: OK

**Comment by mbaechler** [comment]
:+1:

**Comment by rouazana** [comment]
:+1:

**Comment by fvn-linagora** [comment]
:+1: +1

**Comment by aduprat** [comment]
merged

**Pull request closed without merging**